agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -17,8 +17,29 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+
+  /** An enum to represent the different options for availability at every meeting time */
+  private enum Availability { 
+    ALL_AVAILABLE,
+    MANDATORY_AVAILABLE,
+    UNAVAILABLE;
+  
+    boolean isReplaceableBy(Availability availability) {
+      switch (this) {
+        case ALL_AVAILABLE:
+          return true;
+        case UNAVAILABLE:
+          return false;
+        case MANDATORY_AVAILABLE:
+          return availability == ALL_AVAILABLE;
+        default:
```

Comment:
I added the ```default``` case  because I wanted the method to be accurate even if there were other types in the enums. I can just remove this though if you think it's best.

Reply:
Done

Reply:
I'd be concerned that if a new value was added to the enum, it would be easy to overlook updating this, and return an inaccurate value. Perhaps an exception can be thrown in the default case to avoid that?

Reply:
That makes sense. I will change this. Thanks!

Reply:
Done